src/ui/public/filter_editor/lib/filter_editor_utils.js

```diff
@@ -83,7 +83,7 @@ export function buildFilter({ indexPattern, field, operator, params, filterBuild
 }
 
 export function areIndexPatternsProvided(indexPatterns) {
-  if (indexPatterns === undefined || indexPatterns.length === 0 || indexPatterns[0] === undefined) {
+  if (_.compact(indexPatterns).length === 0) {
```

Comment:
Hmm in fact you could just do
`return _.compact(indexPatterns).length === 0;` instead of having the `if` statement.